libraries/botbuilder-dialogs-adaptive-teams/src/actions/sendTaskModuleUrlResponse.ts

```diff
@@ -0,0 +1,114 @@
+    extends BaseSendTaskModuleContinueResponse
+    implements SendTaskModuleUrlResponseConfiguration {
+    /**
+     * Class identifier.
+     */
+    public static $kind = 'Teams.SendTaskModuleUrlResponse';
+
+    /**
+     * Gets or sets an optional expression for the Url of the Task Module response.
+     */
+    public url: StringExpression;
+
+    /**
+     * Gets or sets an optional expression for the Fallback Url the Task Module Task Info response.
+     */
+    public fallbackUrl: StringExpression;
+
+    public getConverter(property: keyof SendTaskModuleUrlResponseConfiguration): Converter | ConverterFactory {
+        switch (property) {
+            case 'disabled':
+                return new BoolExpressionConverter();
+            case 'url':
+            case 'fallbackUrl':
+                return new StringExpressionConverter();
+            default:
+                return super.getConverter(property);
+        }
+    }
+
+    /**
+     * Called when the dialog is started and pushed onto the dialog stack.
+     *
+     * @param {DialogContext} dc The [DialogContext](xref:botbuilder-dialogs.DialogContext) for the current turn of conversation.
+     * @param {object} options Optional, initial information to pass to the dialog.
+     * @returns {Promise<DialogTurnResult>} A promise representing the asynchronous operation.
+     */
+    public async beginDialog(dc: DialogContext, options?: Record<string, unknown>): Promise<DialogTurnResult> {
+        if (this.disabled && this.disabled.getValue(dc.state)) {
+            return dc.endDialog();
+        }
+
+        const url = this.url?.getValue(dc.state);
+        if (!url) {
+            throw new Error(`Missing Url for ${SendTaskModuleUrlResponse.$kind}.`);
+        }
+
+        const title = this.title?.getValue(dc.state);
+        const height = this.height?.getValue(dc.state);
+        const width = this.width?.getValue(dc.state);
+        const completionBotId = this.completionBotId?.getValue(dc.state);
+        const fallbackUrl = this.fallbackUrl?.getValue(dc.state);
+
+        const response = <TaskModuleResponse>{
+            task: <TaskModuleContinueResponse>{
+                value: <TaskModuleTaskInfo>{
+                    height,
+                    width,
+                    title,
+                    completionBotId,
+                    url: encodeURI(url),
+                    fallbackUrl: encodeURI(fallbackUrl),
+                },
+            },
+            cacheInfo: this.getCacheInfo(dc),
+        };
+
+        const responseActivity = BaseTeamsCacheInfoResponseDialog.createInvokeResponseActivity(response);
+        const sendResponse = await dc.context.sendActivity(responseActivity);
+
+        return dc.endDialog(sendResponse);
+    }
+
+    /**
+     * Builds the compute Id for the dialog.
+     *
+     * @returns {string} A string representing the compute Id.
+     */
+    protected onComputeId(): string {
+        return `${this.constructor.name}[
```

Comment:
Using `this.constructor.name` can be problematic if anyone uses a bundler that performs code minification. Is it important that this ID is human readable?

Reply:
I wondered if something like that might cause an issue. My goal was to just make it more dynamic vs. writing it out by hand. Is there a better way?